aten/src/ATen/native/Sorting.cpp

```diff
@@ -182,13 +183,23 @@ void quantile_impl(
   TORCH_CHECK(
       sorted.size(-1) <= std::pow(2, 24),
       "quantile() input tensor is too large");
-  
-  Tensor ranks = q * (sorted.size(-1) - 1);
+
+  Tensor ranks;
+  if (ignore_nan) {
+    ranks = q * (sorted.isnan().logical_not().sum(-1, true) - 1);
+    ranks.masked_fill_(ranks == -1, 0);
+  } else {
+    ranks = q * (sorted.size(-1) - 1);
+  }
+
   Tensor ranks_below = ranks.toType(kLong);
   Tensor weights = ranks - ranks_below;
   Tensor ranks_above = ranks.ceil_().toType(kLong);
-  Tensor values_below = sorted.index_select(-1, ranks_below);
-  Tensor values_above = sorted.index_select(-1, ranks_above);
+
+  Tensor values_below = ignore_nan ? sorted.gather(-1, ranks_below)
```

Comment:
Why gather vs. index_select here?

Reply:
If !ignore_nan, ranks will be 1D which gather does not accept. Gather requires the same number of dimensions as the input.